packages/core/src/utils/rateLimiting/slidingWindowRateLimiter.ts

```diff
@@ -0,0 +1,66 @@
+import { aiLogger } from "@oakai/logger";
+import { Ratelimit } from "@upstash/ratelimit";
+import { waitUntil } from "@vercel/functions";
+import { kv } from "@vercel/kv";
+
+import { RateLimitExceededError } from "./errors";
+import type { Duration, RateLimiter } from "./types";
+
+const log = aiLogger("rate-limiting");
+
+type SlidingWindowRateLimiterArgs = {
```

Comment:
Style question - I normally do export interface. Should we decide on a preferred pattern?

Reply:
I'll update this. I don't know the pros and cons of each approach, but it sounds sensible to me